### ISSUE TITLE
Select filter doesn't fire onChange event when using laptop's touchpad

### ISSUE THREAD

**Issue by ilko-nikolov** [icflorescu/mantine-datatable#595]
Description of what the bug is:
If a table has a filter with a mantine Select/Multiselect component, the filter doesn't trigger the onChange function when an option is selected using a laptop's touchpad. 

0. Have a laptop with touchpad
1. Go to https://icflorescu.github.io/mantine-datatable/examples/searching-and-filtering/
2. Open the filter on "Department name" column. 
3. Tap on any of the options using a laptop's touchpad. 
4. The filter will close without running the onChange function, defined in the Select/MultiSelect component. 

Expected behavior:
After selecting one of the options, the filter and the data on the table should be updated. 

Laptop:
- OS: Windows 10
- Browsers: Edge 124.0.2478.105, Firefox 126.0 

Additional context:
Select without mantine-datatable works like expected - https://mantine.dev/core/select/
I noticed that it does work when you press the touchpad, until it clicks, but just tapping it doesn't work. Also it works on a smartphone or with a keyboard.

**Comment by SeniorCluckers** [comment]
Does Windows have a similar option to mac trackpad?

<img width="485" alt="image" src="https://github.com/icflorescu/mantine-datatable/assets/10767399/13b23bfc-db57-4fb0-8bd6-fdc6837e9040">

If the "tap to click" option is not enabled I have the issue you're describing.

**Comment by ilko-nikolov** [comment]
The behavior should be consistent with https://mantine.dev/core/select/, regardless of the options users may have.

**Comment by tobigumo** [comment]
I encountered the same issue using the mouse.

The `Popover` detects `mousedown` and `touchstart` events by default.
I found that when using a `Select` component inside a `Popover`, if the click or tap lasts even slightly longer, the `Popover` closes prematurely.

It seems this issue can be resolved by changing the events to `mouseup` and `touchend` in the `Popover` options.

- https://mantine.dev/core/popover/#click-outside
- https://codesandbox.io/p/sandbox/mantine-react-template-forked-4pyw2z?file=%2Fsrc%2FApp.tsx%3A20%2C13

Since the `Popover` option used in the filter needs to be adjusted within the `mantine-datatable` package itself, I plan to create a pull request to address this.
